clinica/pydra/t1_volume/t1_volume_utils.py

```diff
@@ -0,0 +1,536 @@
+from os import PathLike
+from typing import Union
+
+from nibabel.nifti1 import Nifti1Header
+from numpy import ndarray
+
+
+def initialize_tissues_spm_segment(
+    tissue_probability_map: Union[PathLike, None] = None
+) -> tuple:
+
+    """Prepare data structure for SPM segment interface
+
+    Parameters
+    ----------
+    tissue_probability_map: Union(PathLike, None)
+        Path to the niftii tissue probability map
+
+    Returns
+    -------
+    tuple
+        "tissues" data structure for SPMSegment
+    """
+
+    import clinica.pydra.t1_volume.t1_volume_utils as spm_utils
+
+    parameters = {}
+    parameters.setdefault("tissue_classes", [1, 2, 3])
+    parameters.setdefault("dartel_tissues", [1, 2, 3])
+    parameters.setdefault("save_warped_unmodulated", True)
+    parameters.setdefault("save_warped_modulated", False)
+
+    if tissue_probability_map:
+        parameters.setdefault("tissue_probability_maps", tissue_probability_map)
+    else:
+        parameters.setdefault("tissue_probability_maps", spm_utils.get_tpm())
+
+    tissue_tuples = get_tissue_tuples(
+        parameters["tissue_probability_maps"],
+        parameters["tissue_classes"],
+        parameters["dartel_tissues"],
+        parameters["save_warped_unmodulated"],
+        parameters["save_warped_modulated"],
+    )
+    return tissue_tuples
+
+
+def init_input_node(bids_name: str) -> None:
+    """Extract "sub-<participant_id>_ses-<session_label>" from <bids_name> and print begin message.
+
+    Parameters
+    ----------
+    bids_name : str
+        The BIDS name of the t1w image
+
+    Warns
+    -----
+    Outputs ID of subjet/session being processed
+    """
+
+    from clinica.utils.filemanip import get_subject_id
+    from clinica.utils.ux import print_begin_image
+
+    subject_id = get_subject_id(bids_name)
+    print_begin_image(subject_id)
+
+    return subject_id, bids_name
+
+
+def print_end_pipeline(bids_name: str) -> None:
+    """Prints end message.
+
+    Parameters
+    ----------
+    bids_name : str
+        The BIDS name of the t1w image
+
+    Warns
+    -----
+    Prints out message for image <bids_name>
+    """
+    from clinica.utils.ux import print_end_image
+
+    print_end_image(bids_name)
+
+
+def zip_list_files(class_images: list, zip_files: bool = False) -> list:
```

Comment:
Same comment as with the `zip_nii` utility function earlier. Sounds that this should be handled with a splitter on a `GZip` task.

Reply:
Good point, I'll look into it.

Reply:
Also not used here. Also removed for now.